Marlin/src/lcd/menu/menu.cpp

```diff
@@ -341,6 +341,45 @@ void scroll_screen(const uint8_t limit, const bool is_menu) {
 
 #endif // BABYSTEP_ZPROBE_OFFSET
 
+#if ENABLED(BABYSTEP_GLOBAL_Z_OFFSET)
+
+  #include "../../feature/babystep.h"
+  #include "../../module/motion.h"
+
+  void lcd_babystep_global_zoffset() {
+    if (!babystep.can_babystep(Z_AXIS)) {
+      // TODO: Needs proper message
+      TERN_(HAS_DISPLAY, ui.set_status(GET_TEXT(MSG_HOME_FIRST)));
+      return ui.return_to_status();
+    }
+
+    if (ui.use_click()) return ui.goto_previous_screen_no_defer();
+    ui.defer_status_screen();
+
+    if (ui.encoderPosition) {
+      int16_t babystep_increment = int16_t(ui.encoderPosition) * (BABYSTEP_SIZE_Z);
+
+      // For large probe offsets moving one step at a time is very slow
+      if (ui.encoderPosition > 2) babystep_increment *= 10;
+
+      const float difference_in_mm = planner.steps_to_mm[Z_AXIS] * babystep_increment,
+                  new_z_offset = home_offset[Z_AXIS] + difference_in_mm;
+
+      ui.encoderPosition = 0;
+      ui.refresh(LCDVIEW_CALL_REDRAW_NEXT);
+
+      //Use difference_in_mm instead of babystep_increment to cover cases where offset was changed by G-code since last UI update
+      babystep.add_mm(Z_AXIS, difference_in_mm);
+      set_home_offset(Z_AXIS, new_z_offset);
```

Comment:
This won't work because it either doubles or cancels out the effect. You can do one or the other, but not both. The `home_offset` (i.e., `workspace_offset`) is applied to every incoming movement command in `GcodeSuite::get_destination_from_command` because these are workspace offsets. They are removed at the earliest opportunity so that all further work can be done in the native workspace.

This is why you cannot both babystep and modify the `position_shift` from `G92` at the same time. The nozzle gets moved, and then also every move command's Z position gets moved. This is why you also can't babystep and modify the mesh Z offset (or the whole mesh) at the same time.

In any case, the `home_offset` is not a friendly option due to its connection to the workspace offset. The option has lost its former utility, so maybe the best idea is to revert to its old behavior. The general problem is to keep this option useful (and, secondarily, to allow it to be modified on the fly) while still always keeping proper track of where the real physical bounds are. In older Marlin you could modify it any time because it was only applied on `G28` and then never used again.